drivers/sub8_videoray_m5_thruster/nodes/thruster_driver.py

```diff
@@ -238,10 +238,27 @@ def thrust_cb(self, msg):
     def alert_thruster_unloss(self, thruster_name):
         if thruster_name in self.failed_thrusters:
             self.failed_thrusters.remove(thruster_name)
-        
         if len(self.failed_thrusters) == 0:
             self.thruster_out_alarm.clear_alarm(parameters={"clear_all"})
         else:
+            rospy.wait_for_service('update_thruster_layout')
+            try:
+                update = rospy.ServiceProxy('update_thruster_layout', UpdateThrusterLayout)
+                rospy.wait_for_service('b_matrix')
+                try:
+                    B = rospy.ServiceProxy('b_matrix', BMatrix)
```

Comment:
I don't think this does what you want. This doesn't call the service, it returns an object to call it later. See [rospy.ServiceProxy](http://wiki.ros.org/rospy/Overview/Services#Service_proxies). Try:
```
# in __init__ so you only have to do this once
rospy.wait_for_service('b_matrix')
self.bmatrix_proxy = rospy.ServiceProxy('b_matrix', BMatrix)
...
#In alert_thruster_unloss
B = self.bmatrix_proxy() #You might have to put an empty BMatirxRequest() as an argument
B = numpy.asmatrix(B.B)
```